src/subdomains/yield-machine/application/services/yield-machine.service.ts

```diff
@@ -0,0 +1,205 @@
+      amount: new BigNumber(parameters.amount),
+      ownerWallet: vault.wallet,
+      accountIndex: vault.accountIndex,
+    });
+  }
+
+  private addPoolLiquidity(vault: Vault, parameters: AddPoolLiquidityParameters): Promise<string> {
+    return this.transactionExecutionService.addPoolLiquidity({
+      from: vault.address,
+      partA: {
+        token: vault.blockchainPairTokenAId,
+        amount: new BigNumber(parameters.partAAmount),
+      },
+      partB: {
+        token: vault.blockchainPairTokenBId,
+        amount: new BigNumber(parameters.partBAmount),
+      },
+      ownerWallet: vault.wallet,
+      accountIndex: vault.accountIndex,
+    });
+  }
+
+  private removePoolLiquidity(vault: Vault, parameters: RemovePoolLiquidityParameters): Promise<string> {
+    return this.transactionExecutionService.removePoolLiquidity({
+      from: vault.address,
+      token: vault.blockchainPairId,
+      amount: new BigNumber(parameters.amount),
+      ownerWallet: vault.wallet,
+      accountIndex: vault.accountIndex,
+    });
+  }
+
+  private compositeSwap(
+    parameters: CompositeSwapParameters,
+    fromTokenId: number,
+    toTokenId: number,
+    ownerWallet: string,
+    accountIndex: number,
+  ): Promise<string> {
+    return this.transactionExecutionService.compositeSwap({
+      source: {
+        from: parameters.address,
+        token: fromTokenId,
+        amount: new BigNumber(parameters.amount),
+      },
+      destination: {
+        token: toTokenId,
+      },
+      ownerWallet,
+      accountIndex,
+    });
+  }
+
+  private checkParameters(command: TransactionCommand, parameters: any): [boolean, boolean] {
+    switch (command) {
+      case TransactionCommand.ACCOUNT_TO_ACCOUNT:
+        return [this.isParameterComplete(parameters, ['from', 'to', 'token', 'amount']), false];
+      case TransactionCommand.CREATE_VAULT:
+        return [this.isParameterComplete(parameters, ['address']), true];
+      case TransactionCommand.DEPOSIT_TO_VAULT:
+        return [this.isParameterComplete(parameters, ['address', 'vault', 'amount']), true];
+      case TransactionCommand.WITHDRAW_FROM_VAULT:
+        return [this.isParameterComplete(parameters, ['address', 'vault', 'amount']), true];
+      case TransactionCommand.TAKE_LOAN:
+        return [this.isParameterComplete(parameters, ['address', 'vault', 'amount']), true];
+      case TransactionCommand.PAYBACK_LOAN:
+        return [this.isParameterComplete(parameters, ['address', 'vault', 'amount']), true];
+      case TransactionCommand.POOL_ADD_LIQUIDITY:
+        return [this.isParameterComplete(parameters, ['address', 'partAAmount', 'partBAmount']), true];
+      case TransactionCommand.POOL_REMOVE_LIQUIDITY:
+        return [this.isParameterComplete(parameters, ['address', 'amount']), true];
+      case TransactionCommand.COMPOSITE_SWAP:
+        return [this.isParameterComplete(parameters, ['address', 'fromToken', 'amount', 'toToken']), true];
+    }
+  }
```

Comment:
This can be done by class validator. Add decorators at `TransactionInputDto`, something like:
```ts
  @ValidateNested()
  @Type(({ object }) => {
    switch ((object as TransactionInputDto).command) {
      case TransactionCommand.ACCOUNT_TO_ACCOUNT:
        return SendTokenParameters;
      ...
    }
  })
```
The interfaces probably need to be changed to classes then.